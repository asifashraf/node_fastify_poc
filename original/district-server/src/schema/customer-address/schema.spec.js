const { fetchGraphQL, getFirstId } = require('../../lib/test-util');
const { customerAddressDetails } = require('../../lib/test-fragments');
const {
  customers: [{ id: customerId }],
} = require('../../../database/seeds/development');

test('customer can resolve defaultAddress', async () => {
  const query = `{
    customer(id: "${customerId}") {
      defaultAddress { ${customerAddressDetails} }
    }
  }`;
  // openInGraphiql(query);
  const result = await fetchGraphQL(query);
  expect(result).toMatchSnapshot();
});

test('customer can resolve addresses', async () => {
  const query = `{
    customer(id: "${customerId}") {
      addresses { ${customerAddressDetails} }
    }
  }`;
  // openInGraphiql(query);
  const result = await fetchGraphQL(query);
  expect(result).toMatchSnapshot();
});

test('customer can save address', async () => {
  const { id: neighborhoodId } = await getFirstId('neighborhoods');
  const query = `mutation{
                   customerSaveAddress(address:{
                    friendlyName: "My Friendly Address"
                    isDefault: true
                    latitude:12.121212
                    longitude:12.4444
                    block:"12"
                    street: "some street"
                    buildingName: "EMPIRE STATE"
                    neighborhoodId:"${neighborhoodId}"
                    streetNumber: "12"
                    type:HOUSE
                    countryCode: "AE"
                    extraFields: [{
                      id: "cc451fc7-6585-4bdd-bcab-c32b20568924"
                      value: "field value - nickname"
                    },
                    {
                      id: "141afd5c-b04e-47cc-b67b-b6b57ac3702c"
                      value: "field value - details"
                    },
                    ]
                  }) {
                     error
                    errors
                    customer {
                      defaultAddress {
                        friendlyName
                        buildingName
                        isDefault
                        id
                        city
                        countryCode
                        extraFields{
                          id
                          name {
                            en
                            ar
                          }
                          value
                          isRequired
                        }
                      }
                      addresses {
                        id
                        isDefault
                        countryCode
                        extraFields{
                          id
                          name {
                            en
                            ar
                          }
                          value
                          isRequired
                        }
                      }
                    }
                   }
                  }`;

  // openInGraphiql(query);
  const result = await fetchGraphQL(query);
  expect(result).toMatchSnapshot();
});
