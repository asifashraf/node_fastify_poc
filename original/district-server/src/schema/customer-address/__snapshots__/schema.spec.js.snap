// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customer can resolve addresses 1`] = `
Object {
  "data": Object {
    "customer": Object {
      "addresses": Array [
        Object {
          "extraFields": Array [
            Object {
              "id": "cc451fc7-6585-4bdd-bcab-c32b20568924",
              "isRequired": false,
              "name": Object {
                "ar": "nickname - ar",
                "en": "nickname",
              },
              "value": "my nickname 2",
            },
            Object {
              "id": "141afd5c-b04e-47cc-b67b-b6b57ac3702c",
              "isRequired": false,
              "name": Object {
                "ar": "details - ar",
                "en": "detalils",
              },
              "value": "my details 2",
            },
          ],
          "friendlyName": "Work",
          "id": "5a7888fe-6d54-48a8-a5f5-fffeaf9e3173",
          "isDefault": false,
          "latitude": 32.3774796,
          "longitude": 44.9903626,
          "neighborhood": Object {
            "id": "62b99864-eebe-4904-b985-824f8bcdff1f",
            "name": Object {
              "ar": "العديلية",
              "en": "Adailiya",
            },
          },
        },
        Object {
          "extraFields": Array [
            Object {
              "id": "cc451fc7-6585-4bdd-bcab-c32b20568924",
              "isRequired": false,
              "name": Object {
                "ar": "nickname - ar",
                "en": "nickname",
              },
              "value": "my nickname 1",
            },
            Object {
              "id": "141afd5c-b04e-47cc-b67b-b6b57ac3702c",
              "isRequired": false,
              "name": Object {
                "ar": "details - ar",
                "en": "detalils",
              },
              "value": "my details 1",
            },
          ],
          "friendlyName": "Home",
          "id": "e4b0f53f-48bc-4f75-a246-9505ab6ee8b3",
          "isDefault": true,
          "latitude": 29.3774796,
          "longitude": 47.9903626,
          "neighborhood": Object {
            "id": "5765880a-aca0-456b-bc1c-9e23cbd5785d",
            "name": Object {
              "ar": "السلام",
              "en": "Salam",
            },
          },
        },
      ],
    },
  },
}
`;

exports[`customer can resolve defaultAddress 1`] = `
Object {
  "data": Object {
    "customer": Object {
      "defaultAddress": Object {
        "extraFields": Array [
          Object {
            "id": "cc451fc7-6585-4bdd-bcab-c32b20568924",
            "isRequired": false,
            "name": Object {
              "ar": "nickname - ar",
              "en": "nickname",
            },
            "value": "my nickname 1",
          },
          Object {
            "id": "141afd5c-b04e-47cc-b67b-b6b57ac3702c",
            "isRequired": false,
            "name": Object {
              "ar": "details - ar",
              "en": "detalils",
            },
            "value": "my details 1",
          },
        ],
        "friendlyName": "Home",
        "id": "e4b0f53f-48bc-4f75-a246-9505ab6ee8b3",
        "isDefault": true,
        "latitude": 29.3774796,
        "longitude": 47.9903626,
        "neighborhood": Object {
          "id": "5765880a-aca0-456b-bc1c-9e23cbd5785d",
          "name": Object {
            "ar": "السلام",
            "en": "Salam",
          },
        },
      },
    },
  },
}
`;

exports[`customer can save address 1`] = `
Object {
  "data": Object {
    "customerSaveAddress": Object {
      "customer": Object {
        "addresses": Array [
          Object {
            "countryCode": "AE",
            "extraFields": Array [],
            "id": "9f03ad20-3fc0-4896-8917-35324e810d58",
            "isDefault": true,
          },
          Object {
            "countryCode": "KW",
            "extraFields": Array [
              Object {
                "id": "cc451fc7-6585-4bdd-bcab-c32b20568924",
                "isRequired": false,
                "name": Object {
                  "ar": "nickname - ar",
                  "en": "nickname",
                },
                "value": "my nickname 2",
              },
              Object {
                "id": "141afd5c-b04e-47cc-b67b-b6b57ac3702c",
                "isRequired": false,
                "name": Object {
                  "ar": "details - ar",
                  "en": "detalils",
                },
                "value": "my details 2",
              },
            ],
            "id": "5a7888fe-6d54-48a8-a5f5-fffeaf9e3173",
            "isDefault": false,
          },
          Object {
            "countryCode": "KW",
            "extraFields": Array [
              Object {
                "id": "cc451fc7-6585-4bdd-bcab-c32b20568924",
                "isRequired": false,
                "name": Object {
                  "ar": "nickname - ar",
                  "en": "nickname",
                },
                "value": "my nickname 1",
              },
              Object {
                "id": "141afd5c-b04e-47cc-b67b-b6b57ac3702c",
                "isRequired": false,
                "name": Object {
                  "ar": "details - ar",
                  "en": "detalils",
                },
                "value": "my details 1",
              },
            ],
            "id": "e4b0f53f-48bc-4f75-a246-9505ab6ee8b3",
            "isDefault": false,
          },
        ],
        "defaultAddress": Object {
          "buildingName": "EMPIRE STATE",
          "city": null,
          "countryCode": "AE",
          "extraFields": Array [],
          "friendlyName": "My Friendly Address",
          "id": "9f03ad20-3fc0-4896-8917-35324e810d58",
          "isDefault": true,
        },
      },
      "error": null,
      "errors": null,
    },
  },
}
`;
